src/js/buffer.js

```diff
@@ -343,6 +343,27 @@ Buffer.prototype.fill = function(value) {
   return this;
 };
 
+
+// Method: Buffer.from()
+// Buffer.from(Array)
+// Buffer.from(string,encoding)
+// Buffer.from(Buffer)
+// Buffer.from(ArrayBuffer)
+function from(value, encoding, length){
+  var arrayBuffer = native.fromArrayBuffer(value,encoding,length);
```

Comment:
I would use early return:
```
if (arrayBuffer) {
  return arrayBuffer;
}
```
I would put this code into the current buffer constructor. There is no point supporting a feature in `Buffer.from` but not supporting it with `new Buffer`.